### PR TITLE
Scheduler

### PR DESCRIPTION
**Pull request by wang-chen**
Add a scheduler for ICP and other algorithms need a loop stop condition.

### CODE REVIEW

**Comment by wang-chen** [comment]
The current implementation is `pypose.module.ReduceToPlateau`, should we move to `pypose.utils.ReduceToPlateau`? Note that this is different from `pypose.optim.scheduler.StopOnPlateau` which can only work for optimizers, this can work for any algorithms that need a loop.

File: pypose/module/icp.py [hunk 1]
```diff
@@ -49,18 +49,25 @@ class ICP(torch.nn.Module):
         >>> target = torch.tensor([[[0.2,      0.1,  0.],
         ...                         [1.1397, 0.442,  0.],
         ...                         [2.0794, 0.7840, 0.]]])
+        >>> scheduler = pp.module.ReduceToPlateau(steps=10, verbose=True)
         >>> icp = pp.module.ICP()
```
Comment (Murphy41) [inline pypose/module/icp.py:53]: icp = pp.module.ICP(scheduler=scheduler)

**Comment by Murphy41** [comment]
> The current implementation is `pypose.module.ReduceToPlateau`, should we move to `pypose.utils.ReduceToPlateau`? Note that this is different from `pypose.optim.scheduler.StopOnPlateau` which can only work for optimizers, this can work for any algorithms that need a loop.

1. Move `pypose.module.ReduceToPlateau` to `pypose.utils.ReduceToPlateau` sounds better to me, and it is a better practice to ensure modularity and prevent circular dependencies.
2. I had a quick look on `pypose.optim.scheduler.StopOnPlateau`, it appears to be an essential component of the optimizations. Are we going to merge these two schedulers? If that's not the case, what do you think about renaming the scheduler for non-optimization modules/functions to something else to prevent confusion?
